Assets/Editor/CreatureLinker.cs

```diff
@@ -0,0 +1,86 @@
+﻿using UnityEngine;
+using UnityEditor;
+using System;
+using System.IO;
+
+public class CreatureLinker : Editor
+{
+    [MenuItem("Tools/Link Creature Scripts")]
+    public static void LinkCreatureScripts()
+    {
+        string rootGeneraPath = "Assets/Creatures/impl";
+        string[] generaDirectories = Directory.GetDirectories(rootGeneraPath);
+
+        foreach (var generaDir in generaDirectories)
+        {
+            string[] unitDirectories = Directory.GetDirectories(generaDir);
+
+            foreach (var unitDir in unitDirectories)
+            {
+                string unitName = Path.GetFileName(unitDir);
```

Comment:
this is a nice extraction, some method extraction might be good

Reply:
Yeah agreed extracted this